rust/src/common/metric_store/influxdb.rs

```diff
@@ -1,48 +1,79 @@
 
-pub enum MetricOwner {
-    Coordinator,
-    Participant,
+pub enum Measurement {
+    Round(RoundMeasurement),
+    Counters(CountersMeasurement),
 }
 
-impl From<&MetricOwner> for &'static str {
-    fn from(metric_owner: &MetricOwner) -> &'static str {
-        match metric_owner {
-            MetricOwner::Coordinator => "coordinator",
-            MetricOwner::Participant => "participant",
+#[derive(InfluxDbWriteable)]
+pub struct RoundMeasurement {
+    time: DateTime<Utc>,
+    round: u32,
+}
+
+impl RoundMeasurement {
+    pub fn new(round: u32) -> RoundMeasurement {
+        RoundMeasurement {
+            time: Timestamp::Now.into(),
+            round,
         }
     }
 }
 
-impl ToString for MetricOwner {
-    fn to_string(&self) -> String {
-        Into::<&str>::into(self).into()
+impl From<RoundMeasurement> for Measurement {
+    fn from(value: RoundMeasurement) -> Self {
+        Self::Round(value)
     }
 }
 
-pub struct Metric(pub MetricOwner, pub Vec<(&'static str, Type)>);
-
-pub struct InfluxDBMetricStore {
-    client: Client,
-    receiver: UnboundedReceiver<Metric>,
+#[derive(InfluxDbWriteable)]
+pub struct CountersMeasurement {
+    time: DateTime<Utc>,
+    number_of_selected_participants: u32,
+    number_of_waiting_participants: u32,
+    number_of_done_participants: u32,
+    number_of_done_inactive_participants: u32,
+    number_of_ignored_participants: u32,
 }
 
-pub async fn run_metricstore(mut influx_client: InfluxDBMetricStore) {
-    loop {
-        match influx_client.receiver.recv().await {
-            Some(Metric(metric_owner, fields)) => {
-                let mut write_query: WriteQuery =
-                    Query::write_query(Timestamp::Now, metric_owner.to_string());
+impl CountersMeasurement {
+    pub fn new(
+        number_of_selected_participants: u32,
+        number_of_waiting_participants: u32,
+        number_of_done_participants: u32,
+        number_of_done_inactive_participants: u32,
+        number_of_ignored_participants: u32,
+    ) -> CountersMeasurement {
+        CountersMeasurement {
+            time: Timestamp::Now.into(),
+            number_of_selected_participants,
+            number_of_waiting_participants,
+            number_of_done_participants,
+            number_of_done_inactive_participants,
+            number_of_ignored_participants,
+        }
+    }
+}
 
-                for (name, value) in fields {
-                    write_query = write_query.add_field(name, value);
-                }
+impl From<CountersMeasurement> for Measurement {
+    fn from(value: CountersMeasurement) -> Self {
+        Self::Counters(value)
+    }
+}
 
-                // Submit the query to InfluxDB.
-                let _ = influx_client
+pub async fn run_metricstore(mut influxdb_connector: InfluxDBConnector) {
+    loop {
+        match influxdb_connector.receiver.recv().await {
+            Some(measurement) => {
+                let query = match measurement {
```

Comment:
As discussed on slack the could be extracted in an impl of [`InfluxDbWriteable`](https://docs.rs/influxdb/0.1.0/influxdb/trait.InfluxDbWriteable.html) for `Measurement` itself:

```rust
impl InfluxDbWriteable for Measurement {
    fn into_query<I: Into<String>>(self, name: I) -> WriteQuery {
        match self {
 		     Self::Round(round) => round.into_query(name),
 		     Self::Counters(counters) => counters.into_query(name)
        }
    }
}
```

I think that would be slightly cleaner, but that's perhaps just a matter of personal opinion so take this as a mere suggestion.